lib/reverse_sentence.rb

```diff
@@ -1,6 +1,41 @@
-# A method to reverse the words in a sentence, in place.
-# Time complexity: ?
-# Space complexity: ?
-def reverse_sentence(my_sentence)
-  raise NotImplementedError
-end
+# Time complexity: 0(n) - no nested loops; each single loop runs the length of the string
+# Space complexity: 0(1) - characters are reversed in place
+def reverse_sentence(my_string)
+  return nil if my_string.nil?
+  return "" if my_string.empty?
+  
+  half = my_string.length / 2
+  j = my_string.length - 1
+  temp_variable = " "
+  
+  # reverses the sentence
+  i = 0
+  while i < half
+    temp_variable = my_string[i]
+    my_string[i] = my_string[j]
+    my_string[j] = temp_variable
+    i += 1
+    j -= 1
+  end
```

Comment:
Notice you're reversing a string or substring twice in this method.  Can you think of a way to abstract this reverse into a helper method:

`def reverse_substring(string, start_index, end_index)`